core/src/typecheck/mod.rs

```diff
@@ -2420,7 +2420,29 @@ pub fn subsumption(
     checked: UnifType,
 ) -> Result<(), UnifError> {
     let inferred_inst = instantiate_foralls(state, &mut ctxt, inferred, ForallInst::UnifVar);
-    checked.unify(inferred_inst, state, &ctxt)
+    match (inferred_inst.clone(), checked.clone()) {
```

Comment:
```suggestion
    match (inferred_inst, checked) {
```

Those early clones are probably not needed. You might need to put a `ref` in the pattern matching though, as in:

```rust
           UnifType::Concrete {
                typ: TypeF::Record(ref rrows),
                ..
            },
```